### PR TITLE
CB-8088 Periscope Downscaling should use YARN downscale recommendation

### PR DESCRIPTION
**Pull request by smaniraju**
Periscope Downscaling Candidates selection is integrated with Yarn Response.
Yarn Scaling Response refactored to be reused between Schedule-Based and Load-Based Autoscaling.

### CODE REVIEW

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/CronTimeEvaluator.java [hunk 1]
```diff
@@ -80,15 +108,21 @@ public void execute() {
...
-    private void publishIfNeeded(List<TimeAlert> alerts) {
+    protected void publishIfNeeded(List<TimeAlert> alerts) {
+        TimeAlert triggeredAlert = null;
         for (TimeAlert alert : alerts) {
-            if (isPolicyAttached(alert) && isTrigger(alert)) {
+            boolean alertTriggerable = isTrigger(alert);
+            if (isPolicyAttached(alert) && alertTriggerable && null == triggeredAlert) {
                 publish(alert);
-                break;
+                triggeredAlert = alert;
```
Comment (sidseth) [inline autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/CronTimeEvaluator.java:121]: All of the history changes here are part of another patch right? Can you please backport that independently of this change?

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnResponseUtils.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+public class YarnResponseUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(YarnResponseUtils.class);
+
+    public List<String> getYarnRecommendedDecommissionHostsForHostGroup(String clusterCrn, YarnScalingServiceV1Response yarnResponse,
+            Map<String, String> hostFqdnsToInstanceId, int maxAllowedDownScale, Optional<Integer> mandatoryDownScaleCount) {
+        Set<String> consideredNodeIds = new HashSet<>();
+        Integer allowedDownscale = Math.max(maxAllowedDownScale, mandatoryDownScaleCount.orElse(0));
+        List<String> decommissionNodes = yarnResponse.getScaleDownCandidates().orElse(List.of()).stream()
+                .sorted(Comparator.comparingInt(YarnScalingServiceV1Response.DecommissionCandidate::getAmCount))
```
Comment (sidseth) [inline autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnResponseUtils.java:30]: Is this needed if YARN is already processing this?

Comment (smaniraju) [reply autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnResponseUtils.java]: removed the sorting.

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/CronTimeEvaluator.java (hunk 1 above)
Comment (smaniraju) [reply autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/CronTimeEvaluator.java]: done.

**Pull request merged**